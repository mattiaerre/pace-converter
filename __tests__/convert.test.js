const convert = require('../src/convert');
const units = require('../src/units');

describe('convert', () => {
  const scenarios = [
    {
      pace: '4:30',
      unit: units.MINKM
    },
    {
      pace: '4:40',
      unit: units.MINKM
    },
    {
      pace: '5:00',
      unit: units.MINKM
    },
    {
      pace: 12,
      unit: units.KMH
    },
    {
      pace: 13.6,
      unit: units.KMH
    },
    {
      pace: 3.96,
      unit: units.KMH
    },
    {
      pace: 6.12,
      unit: units.KMH
    }
  ];

  scenarios.forEach((scenario) => {
    test(`when ${scenario.pace} ${scenario.unit} expect model to match snapshot`, () => {
      const model = convert(scenario.pace, scenario.unit);
      expect(model).toMatchSnapshot();
    });
  });

  test('unknown unit to throw error matching snapshot', () => {
    expect(() => { convert(1000, '$'); }).toThrowErrorMatchingSnapshot();
  });
});
