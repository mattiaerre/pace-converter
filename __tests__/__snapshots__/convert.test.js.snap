// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convert unknown unit to throw error matching snapshot 1`] = `"unknown unit"`;

exports[`convert when 07:30.00 minmi expect model to match snapshot 1`] = `
Object {
  "kmh": 12.874,
  "mih": 8,
  "minkm": "04:39.66",
  "minmi": "07:30.00",
}
`;

exports[`convert when 3.96 kmh expect model to match snapshot 1`] = `
Object {
  "kmh": 3.96,
  "mih": 2.461,
  "minkm": "15:09.12",
  "minmi": "24:22.98",
}
`;

exports[`convert when 4:30 minkm expect model to match snapshot 1`] = `
Object {
  "kmh": 13.333,
  "mih": 8.285,
  "minkm": "04:30.00",
  "minmi": "07:14.52",
}
`;

exports[`convert when 4:40 minkm expect model to match snapshot 1`] = `
Object {
  "kmh": 12.857,
  "mih": 7.989,
  "minkm": "04:40.00",
  "minmi": "07:30.60",
}
`;

exports[`convert when 5:00 minkm expect model to match snapshot 1`] = `
Object {
  "kmh": 12,
  "mih": 7.457,
  "minkm": "05:00.00",
  "minmi": "08:02.76",
}
`;

exports[`convert when 6.12 kmh expect model to match snapshot 1`] = `
Object {
  "kmh": 6.12,
  "mih": 3.803,
  "minkm": "09:48.24",
  "minmi": "15:46.62",
}
`;

exports[`convert when 7:30 minmi expect model to match snapshot 1`] = `
Object {
  "kmh": 12.874,
  "mih": 8,
  "minkm": "04:39.66",
  "minmi": "07:30.00",
}
`;

exports[`convert when 8 mih expect model to match snapshot 1`] = `
Object {
  "kmh": 12.874,
  "mih": 8,
  "minkm": "04:39.66",
  "minmi": "07:30.00",
}
`;

exports[`convert when 8:00 minmi expect model to match snapshot 1`] = `
Object {
  "kmh": 12.07,
  "mih": 7.5,
  "minkm": "04:58.26",
  "minmi": "08:00.00",
}
`;

exports[`convert when 8:30 minmi expect model to match snapshot 1`] = `
Object {
  "kmh": 11.36,
  "mih": 7.059,
  "minkm": "05:16.92",
  "minmi": "08:30.00",
}
`;

exports[`convert when 10 mih expect model to match snapshot 1`] = `
Object {
  "kmh": 16.093,
  "mih": 10,
  "minkm": "03:43.68",
  "minmi": "06:00.00",
}
`;

exports[`convert when 12 kmh expect model to match snapshot 1`] = `
Object {
  "kmh": 12,
  "mih": 7.457,
  "minkm": "05:00.00",
  "minmi": "08:02.76",
}
`;

exports[`convert when 13.6 kmh expect model to match snapshot 1`] = `
Object {
  "kmh": 13.6,
  "mih": 8.451,
  "minkm": "04:24.72",
  "minmi": "07:06.00",
}
`;
